figures/scripts/transcriptomic-overview.R

```diff
@@ -0,0 +1,336 @@
+# J. Taroni for ALSF CCDL 2020
+#
+# Makes a multipanel plot for RNA-seq data display items: UMAP, GSVA scores,
+# and immune deconvolution scores (xCell). This only handles stranded data right
+# now.
+
+library(tidyverse)
+library(ComplexHeatmap)
+library(multipanelfigure)
+
+# Establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Declare output directory
+output_dir <- file.path(root_dir, "figures", "pngs")
+
+# Final figure filename
+output_png <- file.path(output_dir, "transcriptomic-overview.png")
+
+# Data directory
+data_dir <- file.path(root_dir, "data")
+
+# Analysis directory
+analyses_dir <- file.path(root_dir, "analyses")
+
+# Read in clinical data
+histologies_df <- read_tsv(file.path(data_dir, "pbta-histologies.tsv"),
+                           guess_max = 10000)
+
+#### Temporary PNG names -------------------------------------------------------
+# Because we are mixing Heatmaps and ggplots in this figure, one of the best
+# ways to control sizes is to create PNGs and then use multipanelfigure to
+# put the final figure together.
+#
+# Here we'll set up the file names used throughout
+umap_png <- file.path(output_dir, "temp_umap.png")
+gsva_png <- file.path(output_dir, "temp_gsva.png")
+deconv_png <- file.path(output_dir, "temp_deconv.png")
+legend_png <- file.path(output_dir, "temp_legend.png")
+
+#### Color palettes ------------------------------------------------------------
+
+palette_dir <- file.path(root_dir, "figures", "palettes")
+histology_palette <- read_tsv(file.path(palette_dir,
+                                               "histology_color_palette.tsv"))
+divergent_palette <- read_tsv(file.path(palette_dir,
+                                               "divergent_color_palette.tsv"))
+gradient_palette <- read_tsv(file.path(palette_dir,
+                                              "gradient_color_palette.tsv"))
+
+#### UMAP plot -----------------------------------------------------------------
+
+dim_red_dir <- file.path(
+  analyses_dir,
+  "transcriptomic-dimension-reduction"
+)
+
+source(file.path(dim_red_dir, "util", "dimension-reduction-functions.R"))
```

Comment:
The convention I'm using here is to call the directories, files, etc. relevant to each panel within that panels section. To do what you suggest, I'd have to move all of this out of the individual panel sections. Normally I would agree with you, but I think in this case having it arranged by panel is helpful in case you want to rip out a panel entirely.